packages/core/src/stateUtils.ts

```diff
@@ -1461,7 +1461,7 @@ interface BuiltinAction {
   execute: (actorScope: AnyActorScope, params: unknown) => void;
 }
 
-function resolveActionsAndContextWorker(
+function resolveAndExecuteActionsWithContext(
```

Comment:
nit: I don't understand the new name - what does the `withContext` bit refer to? 🤔

Reply:
Execute the actions with the latest context - just made the name more clear

Reply:
I still don't get it 😅 I'm not sure what is the context that is provided here - or if this refers to the `snapshot.context` or what. The `*Worker` convention from some other codebase cause I couldn't figure out a specific name for it. I kinda wanted to name 2 functions the same way - the `*Worker` was just a helper function.

I'm open to renaming this since the old name was definitely not perfect but the `*WithContext` confuses me here (especially given that we have `snapshot.context` and I don't see how that fits here).

Reply:
I think I misread the new name slightly at the beginning... but now I find the new name even more inaccurate :D this function *does* resolve both actions and context. I think I see how it could be named what you have named it but at that moment the "outer" call should also be somehow renamed. I don't see what makes this one to be called `resolveAndExecuteActionsWithContext` while the other one is called `resolveActionsAndContext`.